### ISSUE TITLE
404 error accessing results

### ISSUE THREAD

**Issue by mattclay** [landscapeio/landscape-issues#273]
I'm seeing HTTP 404 errors trying to access results:

Starting here: https://landscape.io/github/ansible/ansible/3

Then click on Errors, Smells, Style or All Messages:

- https://landscape.io/github/ansible/ansible/3/messages/error
- https://landscape.io/github/ansible/ansible/3/messages/style
- https://landscape.io/github/ansible/ansible/3/messages/smell
- https://landscape.io/github/ansible/ansible/3/messages

All of those return nginx 404 error pages.

**Comment by carlio** [comment]
@mattclay I don't see those 404 errors myself now, do you still get them? I see the 404 in the nginx logs but no errors or reason for it, but I also just restarted the webapp for a redeploy and perhaps that fixed this somehow. I'll keep poking.

**Comment by mattclay** [comment]
@carlio I reloaded those pages, and all but messages came up fine (it gave me a 404). I then reloaded the pages again and got 404 errors for all but style and smell (even the main page gave me a 404 error).

It's acting like there's multiple load balanced servers, and one (but not all) of the servers are returning 404 errors. I reloaded the pages several more times, and continued to get mostly 404 errors, although I also got a 504 gateway timeout for the main page.

**Comment by carlio** [comment]
@mattclay Aha I figured it out - turns out the 404 is supposed to be a 502. As it turns out, the webserver was running low on memory. However the 502.html page was in the wrong place, which means the 404 you saw was actually "could not find the 502 html". That's why I was confused! I moved the webapp to a new server so this should go away now (once DNS propagates), and I'll look into why it was running out of RAM in the first place too. By restarting the webapp I obviously killed the threads that were hogging memory so accidentally fixed this but only temporarily.

Edit: actually new web server will take a while due to https://github.com/pypa/setuptools/issues/937
